src/Symfony/Component/HttpKernel/DependencyInjection/AddClassesToCachePass.php

```diff
@@ -39,12 +39,27 @@ public function process(ContainerBuilder $container)
         $classes = array();
```

Comment:
What about `$classes = $this->kernel->getClassesToCompile();` and drop all foreaches below.. (ie. is replacing the `array_merge` with `foreach` really needed for performance?)